carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/utils/android/AndroidService.java

```diff
@@ -1056,7 +1056,8 @@ public boolean setDeviceTimeZone(String timeZone, String settingsTZ, TimeFormat
 
         String currentAndroidVersion = DevicePool.getDevice().getOsVersion();
         LOGGER.info("currentAndroidVersion=" + currentAndroidVersion);
-        if (currentAndroidVersion.contains("7.") || (DevicePool.getDevice().getDeviceType() == DeviceType.Type.ANDROID_TABLET)) {
+        if (currentAndroidVersion.contains("7.") || (currentAndroidVersion.contains("7.") &&
```

Comment:
what's the difference in "or " statement?

Reply:
original was as follows:
if device is 7.x or deviceType is Tablet, do TimeZone_changer app, I changed it to:
if device is 7.x or (device is 7.x && deviceType is Tablet).. meaning that if deviceType is tablet and it's 8.x, the original way would execute TimeZone_changer app which wouldn't work, since for 8.x we want to execute "Settings" approach